projects/core/directives/hint/hint-driver.directive.ts

```diff
@@ -0,0 +1,9 @@
+import {Directive} from '@angular/core';
+import {TuiDestroyService} from '@taiga-ui/cdk';
+import {AbstractTuiDriverDirective} from '@taiga-ui/core/abstract';
+
+@Directive({
+    selector: `[tuiHint]`,
+    providers: [TuiDestroyService],
+})
+export class TuiHintDriverDirective extends AbstractTuiDriverDirective {}
```

Comment:
Can we explicitly extends this class by all "drivers" ?

It was difficult for me to realize when we subscribe to all "drivers"-observables :(
Now I understand that it happens here but it not obvious.

Reply:
The idea is we have a driver and we have a vehicle. To connect them together we need to inject and subscribe. And to initiate this action we need a directive with a particular selector. But the action is always the same so I moved it into abstract directive. Basically, for every pair of driver/vehicle we will have a directive like that which is only necessary for its selector and the work it does is the same and happens in abstract class.